lib/engine/game/g_system18/game.rb

```diff
@@ -524,22 +525,28 @@ def emergency_issuable_bundles(entity)
           return [] if game_capitalization != :incremental
           return [] if entity.trains.any?
           return [] unless @depot.min_depot_train
+          return [] if @round.emergency_issued
 
           min_train_price = @depot.min_depot_price
           return [] if entity.cash >= min_train_price
 
-          @corporations.flat_map do |corp|
+          issuable_bundles = @corporations.flat_map do |corp|
             bundles = bundles_for_corporation(entity, corp)
             bundles.select! { |b| @share_pool.fit_in_bank?(b) }
-
-            # Cannot issue more shares than needed to buy the train from the bank
-            train_buying_bundles = bundles.select { |b| (entity.cash + b.price) >= min_train_price }
-            if train_buying_bundles.size > 1
-              excess_bundles = train_buying_bundles[1..-1]
-              bundles -= excess_bundles
-            end
             bundles
           end.compact
+
+          # Sort bundles by price in ascending order
+          issuable_bundles.sort_by!(&:price)
+
+          # Find the smallest bundle that raises enough cash to buy the train
+          issuable_bundles.each do |bundle|
+            total_cash = bundle.price + entity.cash
+            return [bundle] if total_cash >= min_train_price
+          end
+
+          # If no single bundle is enough, return the largest bundle
+          [issuable_bundles.last]
```

Comment:
`issuable_bundles` _could_ contain share bundles for more than one corporation. I guess this is for an 1841-style game where corporations can own shares of other corporations, and these could be issued in emergency money raising.

This is going to return a single bundle for one corporation. It should return a single bundle for each corporation. So this bit needs to be moved into the `@corporations.flat_map` block.

I don't know if any currently implemented rules for System18 allow corporations to own other corporations' shares, but this has been written to allow this, so should be preserved.